hydroflow/src/scheduled/state.rs

```diff
@@ -0,0 +1,18 @@
+use std::marker::PhantomData;
+
+use super::StateId;
+
+#[must_use]
+pub struct StateHandle<T> {
+    pub(crate) state_id: StateId,
+    pub(crate) _phantom: PhantomData<fn() -> T>,
```

Comment:
is there a reason for this to be `fn() -> T` instead of just `T`?

Reply:
I discovered a while back that there's actually a lot of different ways to use `PhantomData` here: https://doc.rust-lang.org/nomicon/phantom-data.html#table-of-phantomdata-patterns

Which now that I actually think about it, should probably be `*mut T` instead of `fn() -> T` lol.. But it's all kinda not actually important